server/index.js

```diff
@@ -47,6 +47,78 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const select = `
+        select *
+        from carts
+  `;
+  db.query(select)
+    .then(result => res.json(result.json))
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId, 10);
+  if (!productId || productId < 1) {
+    return next(new ClientError('ProductId must be a positive interger', 400));
+  }
+  const select = `
+        select "price"
+        from "products"
+        where "productId" =$1
+  `;
+  db.query(select, [productId])
+    .then(result => {
+      if (!result.rows[0]) throw new ClientError('ProductId does not exist', 404);
+      const price = result.rows[0].price;
+      const insert = `
+          insert into "carts" ("cartId", "createdAt")
+          values (default, default)
+          returning "cartId"
+    `;
+      return (
+        db.query(insert)
+          .then(result => {
+            const cartId = result.rows[0].cartId;
+            const object = {
+              cartId: cartId,
+              price: price
+            };
+            return object;
+          })
+      );
```

Comment:
We don't want to create a cart if there is already a cart in the current session.